server-spi-private/src/main/java/org/keycloak/models/locking/LockingMechanismProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.locking;
+
+import org.keycloak.provider.Provider;
+
+public interface LockingMechanismProvider extends Provider {
```

Comment:
Is this a _global lock provider_ rather than a _mechanism_?

Reply:
> Yes, this is needed in exceptional case when Keycloak leaves a lock record. I'd even incline to renaming this method to `unlockAllLockForcibly`.

Do we want to use the words `acquire`/`release` or `lock`/`unlock`?
`lock()`
`unlock()`
`unclokAllLocksForcibly`
vs
`acquireLock`
`releaseLock`
`releaseAllLocksForcibly`
?